migrate/v0_11/migrate.go

```diff
@@ -65,3 +67,53 @@ func MigrateBep3(oldGenState v0_9bep3.GenesisState) v0_11bep3.GenesisState {
 		PreviousBlockTime: v0_11bep3.DefaultPreviousBlockTime,
 	}
 }
+
+// MigrateCDP migrates from a v0.9 (or v0.10) cdp genesis state to a v0.11 cdp genesis state
+func MigrateCDP(oldGenState v0_9cdp.GenesisState) v0_11cdp.GenesisState {
+	var newCDPs v0_11cdp.CDPs
+	var newDeposits v0_11cdp.Deposits
+	var newCollateralParams v0_11cdp.CollateralParams
+	newStartingID := uint64(0)
+
+	for _, cdp := range oldGenState.CDPs {
+		newCDP := v0_11cdp.NewCDPWithFees(cdp.ID, cdp.Owner, cdp.Collateral, "bnb-a", cdp.Principal, cdp.AccumulatedFees, cdp.FeesUpdated)
+		newCDPs = append(newCDPs, newCDP)
+		if cdp.ID >= newStartingID {
+			newStartingID = cdp.ID + 1
```

Comment:
If CDP ID 1005 is the most recently created CDP on kava-3 but it has been closed or liquidated (making ID 1004 the largest seen ID) then on kava-4 the first new CDP will share ID 1005 with a closed CDP from kava-3. Could this cause issues when querying state across kava-3/kava-4?

If we can't check event records here then this may be the most reasonable implementation though.

Reply:
Not sure why I wrote it this way - we have access to the value directly in `oldGenState.StartingCdpID`.